src/helpers/promptUtils.ts

```diff
@@ -0,0 +1,69 @@
+import { Question, Questions, ux } from '@cto.ai/sdk'
+
+//This re-prompts user until their answer passes validation
+//validate takes in the return value of ux.prompt, and forces re-prompt if it returns false
+//errorMessage is displayed to user when validation fails
+export const validatedPrompt = async (
+  prompt: Question | Questions,
+  validate: (response: any) => boolean,
+  errorMessage: string,
+) => {
+  let response: any
+  let repeatedPrompt = false
+  //keep asking as long as validation fails
+  do {
+    if (repeatedPrompt) {
+      //not our first prompt
+      //must have errored, so print the error message
+      await ux.print(errorMessage)
+    }
+    response = await ux.prompt(prompt)
+    repeatedPrompt = true
+  } while (!validate(response))
+  return response
+}
+
+//this replicates the SDK 1.x functionality,
+//where you can have a separate display name and actual value returned
+//prompt should be a ListQuestion, AutoCompleteQuestion, CheckboxQuestion
+//name must be unique, as it is used as a key
+//having duplicate names causes undefined behaviour (probably just returns one of the matching at random)
+//prompt.choices will be overwritten, so feel free to set it to anything
+export const keyValPrompt = async (
+  prompt: Question,
+  choices: { name: string; value: any }[],
+) => {
+  //the ListQuestion interface is not actually exported
+  if (['list', 'autocomplete', 'checkbox'].includes(prompt.name)) {
+    throw `prompt must be one of [list, autocomplete, checkbox], but got ${prompt.name}!`
+  }
+
+  let nameList: string[] = new Array(choices.length)
```

Comment:
Can we find a better name for this? Maybe `choicesArr`? So that it's easy to follow with the conversion from object to array..

Reply:
Got it, replaced with `choicesArr`!